lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1535,6 +1572,20 @@ class SemanticsObject {
       primaryRole = currentPrimaryRole;
       currentPrimaryRole.update();
     }
+
+    // Reparent element.
+    if (currentElement != element) {
+      final DomElement? container = _childContainerElement;
+      if (container != null) {
+        container.remove();
```

Comment:
You don't need to remove the container. The browser will do it automatically on the next line when you call `element.append`.